napari/layers/labels/labels.py

```diff
@@ -312,23 +307,33 @@ def mode(self, mode: Union[str, Mode]):
         if mode == self._mode:
             return
 
+        if self._mode == Mode.PICK:
+            self.mouse_drag_callbacks.remove(pick)
+        elif self._mode == Mode.PAINT:
+            self.mouse_drag_callbacks.remove(paint)
+        elif self._mode == Mode.FILL:
+            self.mouse_drag_callbacks.remove(fill)
+
         if mode == Mode.PAN_ZOOM:
             self.cursor = 'standard'
             self.interactive = True
             self.help = 'enter paint or fill mode to edit labels'
-        elif mode == Mode.PICKER:
+        elif mode == Mode.PICK:
             self.cursor = 'cross'
             self.interactive = False
             self.help = 'hold <space> to pan/zoom, click to pick a label'
+            self.mouse_drag_callbacks.append(pick)
         elif mode == Mode.PAINT:
             self.cursor_size = self.brush_size / self.scale_factor
             self.cursor = 'square'
             self.interactive = False
             self.help = 'hold <space> to pan/zoom, drag to paint a label'
+            self.mouse_drag_callbacks.append(paint)
         elif mode == Mode.FILL:
             self.cursor = 'cross'
             self.interactive = False
             self.help = 'hold <space> to pan/zoom, click to fill a label'
+            self.mouse_drag_callbacks.append(fill)
```

Comment:
Alternatively, a dictionary mapping modes to callbacks could be added. `self.mouse_drag_callbacks.append(callback_dict[mode])` at the very end, without all the repetitive calls in the `elif` cascade. You could also use `self.mouse_drag_callbacks.remove(callback_dict[self._mode])` at the top.

Reply:
This will change in #1048 too, so I'd probably rather just leave, but I can do this if needed